doc/modules/ensemble.rst

```diff
@@ -1158,3 +1157,87 @@ The following example shows how to fit the VotingRegressor::
+are stacked together. It should be given as a list of names and estimators::
+
+  >>> from sklearn.linear_model import RidgeCV, LassoCV
+  >>> from sklearn.svm import SVR
+  >>> estimators = [('ridge', RidgeCV()),
+  ...               ('lasso', LassoCV(random_state=42)),
+  ...               ('svr', SVR(C=1, gamma=1e-6, kernel='rbf'))]
+
+The `final_estimator` will use the predictions of the `estimators` as input. It
+needs to be a classifier or a regressor when using :class:`StackingClassifier`
+or :class:`StackingRegressor`, respectively::
+
+  >>> from sklearn.ensemble import GradientBoostingRegressor
+  >>> from sklearn.ensemble import StackingRegressor
+  >>> reg = StackingRegressor(
+  ...     estimators=estimators,
+  ...     final_estimator=GradientBoostingRegressor(random_state=42))
+
+To train the `estimators` and `final_estimator`, the `fit` method needs
+to be called on the training data::
+
+  >>> from sklearn.datasets import load_boston
+  >>> X, y = load_boston(return_X_y=True)
+  >>> from sklearn.model_selection import train_test_split
+  >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+  ...                                                     random_state=42)
+  >>> reg.fit(X_train, y_train)
+  StackingRegressor(...)
+
+During training, the `estimators` are fitted on the whole training data
+`X_train`. They will be used when calling `predict`, `predict`, or
+`predict_proba`. To generalize and avoid over-fitting, the
+`final_estimator` is trained on out-samples using internally
+:func:`sklearn.model_selection.cross_val_predict`.
+
+Note that the output of the `estimators` is controlled by the parameter
+`predict_method`. It corresponds to the method called by each
+estimator. This parameter is either a list of strings, being method names, or
+`'auto'` which will automatically identify an available method depending on the
+availability and a pre-determined order of preference (`predict_proba`,
+`decision_function` and `predict`).
+
+The fitted stacking estimators have the `predict` method, and classifier also
+have the `predict_proba` method. They can be used as any other
+estimator, e.g.::
+
+   >>> y_pred = reg.predict(X_test)
+   >>> from sklearn.metrics import r2_score
+   >>> print('R2 score: {:.2f}'.format(r2_score(y_test, y_pred)))
+   R2 score: 0.81
+
+Note that it is also possible to get the output of the stacked outputs of the
+`estimators` using the `transform` method::
+
+  >>> reg.transform(X_test[:5])
+  array([[28.78..., 28.43...  , 22.62...],
+         [35.96..., 32.58..., 23.68...],
+         [14.97..., 14.05..., 16.45...],
+         [25.19..., 25.54..., 22.92...],
+         [18.93..., 19.26..., 17.03... ]])
+
+.. note::
+   Multiple stacking layers can be achieved by assigning `final_estimator` to
+   a :class:`StackingClassifier` or :class:`StackingRegressor`.
+
+.. topic:: References
+
+   .. [W1992] Wolpert, David H. "Stacked generalization." Neural networks 5.2
```

Comment:
I'll check tomorrow but I am unsure if there is a stacking part in ESL.

Reply:
8.8 Model Averaging and Stacking (begins on pg. 288 in ESL). So pumped for this!

Reply:
might be for a single model only, though? But definitely more readable than wolpert.

Reply:
@jmwoloso Thanks, this is going faster like this :)

Reply:
@glemaitre no problem! really looking forward to this. let me know how else I can assist.

Reply:
i've also run across the following sources where stacking is discussed, but haven't read them myself, but rather came across them in the vtreat paper (quick read and really awesome stuff as well; they also just released [pyvtreat](https://github.com/WinVector/pyvtreat)). links below if needed/useful. seems i've read before over the years that there are multiple ways to do stacking. it was always on my to-do list to become more familiar with the details. this will be a good excuse :D

[vtreat paper](https://arxiv.org/pdf/1611.09477.pdf)

[super learner in prediction](https://biostats.bepress.com/ucbbiostat/paper266/)

[on cross validation and stacking](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.188.2952&rep=rep1&type=pdf)

[general oracle inequalities for model selection](https://projecteuclid.org/euclid.ejs/1236089916)